discovery-infra/download_logs.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+
+import os
+import json
+from glob import glob
+from argparse import ArgumentParser
+from datetime import date
+
+from logger import log
+from utils import recreate_folder
+from assisted_service_api import create_client, InventoryClient
+
+FAILED_STATUSES = ["error"]
+
+
+def main():
+    args = handle_arguments()
+    client = create_client(url=args.inventory_url)
+
+    if args.cluster_id:
+        cluster = client.cluster_get(args.cluster_id)
+        download_logs(client, cluster, args.dest)
+    else:
+        clusters = client.clusters_list()
+
+        if not clusters:
+            log.info('No clusters were found')
+            return
+
+        for cluster in clusters:
+            if should_download_logs(cluster):
+                download_logs(client, cluster, args.dest)
+
+
+def should_download_logs(cluster: dict):
+    return cluster['status'] in FAILED_STATUSES
+
+
+def download_logs(client: InventoryClient, cluster: dict, dest: str):
+    if glob(f"{dest}/*{cluster['id']}"):
+        log.info(f"Skipping. The logs of cluster {cluster['id']} already exist under '{dest}' directory.")
+        return
+
+    output_folder = os.path.join(dest, f"{date.today()}_{cluster['id']}")
+    recreate_folder(output_folder)
+
+    write_metadata_file(client, cluster, os.path.join(output_folder, 'metdata.json'))
+    client.download_cluster_events(cluster['id'], os.path.join(output_folder, f"cluster_{cluster['id']}_events.json"))
```

Comment:
i am sure i asked this question already. Don't u want to download logs if events fails?